src/OSPSuite.SimModel/src/Species.cpp

```diff
@@ -267,6 +267,9 @@ void Species::RescaleValues ()
 	
 	for(int i =0; i<_valuesSize;i++)
 		_values[i] *= m_ODEScaleFactor;
+
+	//threshold for numeric comparisons must be rescaled the same way!
+	_comparisonThreshold *= m_ODEScaleFactor;
```

Comment:
>Are we sure that SetComparisonThreshold is called BEFORE RescaleValues?

Per construction. This is called at the end of ODE solving routine and before the values are rescaled